### DISCUSSION TITLE
Support for Steam Guard Mobile Authenticator

### DISCUSSION THREAD

**Discussion by rexkyng** [ente-io/ente#1038] [Enhancements]
Add support for Steam Guard Mobile Authenticator, supporting `otpauth://totp/Steam:{yourUserName}?secret{yourTOTPSecret}&issuer=Steam` URI's
The URI can be generated with [steamctl](https://github.com/ValvePython/steamctl), seen in [here](https://github.com/beemdevelopment/Aegis/wiki/Adding-Steam-to-Aegis-from-steamguard-cli)

This feature is implemented in both aegis and andotp, would be great to also get it supported in ente auth.

Thanks.

**Comment by rexkyng** [comment]
related issue on andotp:
https://github.com/andOTP/andOTP/issues/510#issue-589614473

**Comment by T0xIcB3ast** [comment]
Are there any plans to include this?

**Comment by yash-garg** [comment]
Anyone planning to implement this? I'd like to give this a try in my free time.

Just to add [Bitwarden](https://bitwarden.com) also supports Steam TOTP now.

Update: Made some progress [here](https://github.com/ente-io/ente/compare/main...Yash-Garg:ente:main?w=1) but will have to completely generate the **Steam TOTP** separately since the otp package doesn't support it directly.

**Reply by HowieHz** [reply]
I know you can do it.❤️
It would be great if you could make it happen.

**Reply by vishnukvmd** [reply]
Thanks for the inputs @Yash-Garg!

This has been implemented: https://github.com/ente-io/ente/pull/1539/files and will be shipped in our next release.

**Reply by DaFlowah** [reply]
Hi @vishnukvmd,
I can see this is now live in latest release.
Are you gonna add a page on https://help.ente.io with the steps to follow? Or would this be the same as in [here](https://github.com/beemdevelopment/Aegis/wiki/Adding-Steam-to-Aegis-from-steamguard-cli)?

**Reply by vishnukvmd** [reply]
Someone in our Discord community also shared https://www.reddit.com/r/Bitwarden/comments/a67c1n/steam_authenticator_supported_not_sure_how_to/ebunt81/

**Reply by yash-garg** [reply]
> Someone in our Discord community also shared https://www.reddit.com/r/Bitwarden/comments/a67c1n/steam_authenticator_supported_not_sure_how_to/ebunt81/

There's a cli method also incase that one doesn't work - https://github.com/dyc3/steamguard-cli, mostly similar procedure which generates the `maFiles`.

For [Nix](https://nixos.org/) users they can just run:

```bash
nix run nixpkgs#steamguard-cli -- setup
```
